### PR TITLE
Detect Adjust WalletConnected Safes

### PR DESCRIPTION
**Pull request by cag**
Related #127

### CODE REVIEW

File: src/utils/checkConnectedToSafe.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  const wc =
+    (await provider.getWalletConnector?.()) ||
+    (await provider.connection?.getWalletConnector?.()) ||
+    provider.wc ||
+    provider.connection?.wc
+
+  const peerName = wc?.peerMeta?.name
+
+  if (peerName === 'Safe Multisig WalletConnect' || peerName?.startsWith?.('Gnosis Safe')) {
```
Comment (rmeissner) [inline src/utils/checkConnectedToSafe.ts:12]: Should we change this to `startsWith('Safe')` ... just to be on the safe side (no pun intended)

Comment (rmeissner) [reply src/utils/checkConnectedToSafe.ts:12]: Btw the multisig identifies itself as `Gnosis Safe Multisig` so the current logic should work.

see https://github.com/gnosis/safe-react-apps/pull/78/files#diff-791a7865e89890c3f6524eff6f43a645c54438fec18d7064aaff481efa0b9debL27

**Comment by germartinez** [comment]
Do you know any open sourced Dapp that I can use to test this (updating its CPK version with this branch locally)?
I've been trying to make it work in the [CPK Config Dapp](https://gnosis.github.io/cpk-configuration-app/) updating the CPK version with this branch with no success.
I've tried `web3` and `ethers` providers and `web3modal` and `web3Connect` but I'm getting:

`TypeError: this.signer.provider.send is not a function` when using `ethers`
`TypeError: this.send is not a function` when using `web3`

**Comment by germartinez** [comment]
Forget about my previous comment.

Web3Modal has an issue when using web3 1.3.0 (thanks Richard!)
https://ethereum.stackexchange.com/questions/87872/issue-about-connecting-with-walletconnect-by-using-the-example-of-web3modal

I downgraded the version in the CPK Dapp to test it and this PR works

**Review by germartinez — APPROVED** [review]

File: src/utils/checkConnectedToSafe.ts (hunk 1 above)
Comment (cag) [reply src/utils/checkConnectedToSafe.ts:12]: Ah yeah, maybe, though German just merged it with the stricter check, so I guess that's going to stay for now :sweat_smile:

**Pull request merged**